EngineLayer/CompactToProteinMatchEngine/SequencesToActualProteinPeptidesEngine.cs

```diff
@@ -50,63 +52,82 @@ public SequencesToActualProteinPeptidesEngine(List<Psm>[] allPsms, List<Protein>
+                    lock (obj)
                     {
-                        foreach (var peptideWithSetModifications in peptideWithPossibleModifications.GetPeptidesWithSetModifications(variableModifications, maxModificationIsoforms, max_mods_for_peptide))
+                        foreach (var ye in local)
                         {
                             HashSet<PeptideWithSetModifications> v;
-                            if (local.TryGetValue(new CompactPeptide(peptideWithSetModifications), out v))
-                                v.Add(peptideWithSetModifications);
+                            if (compactPeptideToProteinPeptideMatching.TryGetValue(ye.Key, out v))
+                                foreach (var huh in ye.Value)
+                                    v.Add(huh);
+                        }
+                        proteinsSeen += fff.Item2 - fff.Item1;
+                        var new_progress = (int)((double)proteinsSeen / (totalProteins) * 100);
+                        if (new_progress > old_progress)
+                        {
+                            //ReportProgress(new ProgressEventArgs(new_progress, "In adding possible sources to peptide dictionary loop", nestedIds));
+                            old_progress = new_progress;
                         }
                     }
-                lock (obj)
-                {
-                    foreach (var ye in local)
-                    {
-                        HashSet<PeptideWithSetModifications> v;
-                        if (compactPeptideToProteinPeptideMatching.TryGetValue(ye.Key, out v))
-                            foreach (var huh in ye.Value)
-                                v.Add(huh);
-                    }
-                    proteinsSeen += fff.Item2 - fff.Item1;
-                    var new_progress = (int)((double)proteinsSeen / (totalProteins) * 100);
-                    if (new_progress > old_progress)
-                    {
-                        //ReportProgress(new ProgressEventArgs(new_progress, "In adding possible sources to peptide dictionary loop", nestedIds));
-                        old_progress = new_progress;
-                    }
-                }
-            });
+                });
+
+                #endregion Match Seqeunces to PeptideWithSetModifications
 
-            #endregion Match Seqeunces to PeptideWithSetModifications
+                return new SequencesToActualProteinPeptidesEngineResults(this, compactPeptideToProteinPeptideMatching);
+            }
+            else
+            {
+                NonSpecificEnzymeSearch.NonSpecificEnzymeSequencesToActualPeptides NSESTAP = new NonSpecificEnzymeSearch.NonSpecificEnzymeSequencesToActualPeptides(allPsms, proteinList, massDiffAcceptors, protease, maxMissedCleavages, minPeptideLength, maxPeptideLength, initiatorMethionineBehavior, fixedModifications, variableModifications, maxModificationIsoforms, nestedIds, addCompIons, lp);
```

Comment:
Since you are already creating a new engine, just call it from the search task, instead of having the if statement here. This way you would not need to pass in the addCompIons and lp parameters to the original SequencesToActualProteinPeptidesEngine